node/node.go

```diff
@@ -164,6 +182,12 @@ type NodeController struct { // nolint: golint
 	nodeStatusUpdateErrorHandler ErrorHandler
 
 	chReady chan struct{}
+
+	// What was the last error we got from the node upon ping
+	lastPingErrorLock  sync.RWMutex
```

Comment:
There's only one reader and one writer for this, I think?
Seems like they are mutually exclusive anyway and can use a normal `sync.Mutex`